backend/inspirehep/records/api/mixins.py

```diff
@@ -332,18 +328,6 @@ def update_self_citations(self):
             )
         ).update({RecordCitations.is_self_citation: False}, synchronize_session=False)
 
-    def update(self, data, disable_relations_update=False, *args, **kwargs):
-        super().update(data, disable_relations_update, *args, **kwargs)
-
-        if disable_relations_update:
-            LOGGER.info(
-                "Record citation update disabled",
-                recid=self.get("control_number"),
-                uuid=str(self.id),
-            )
-        else:
-            self.update_refs_in_citation_table()
-
```

Comment:
Was this an issue? As you removed this you added separate update for DataRecord class. This code is now duplicated in LiteratureRecord and in DataRecord. And actually it's error prone (as we can see). IMHO you should add `create` modethod here which calls `super().create(data)` and then `self.update_refs_in_citation_table` acordingly. You won't duplicate identical code then in two classes. You have already Mixin, use it.

Reply:
This is a very valid point, but I would like to have a full control in the literature class and call the relations update manually.